client/lib/post-normalizer/rule-wait-for-images-to-load.js

```diff
@@ -43,7 +43,12 @@ function promiseForImage( image ) {
 	} );
 }
 
-const promiseForURL = flow( imageForURL, promiseForImage );
+const promiseForURL = function( urlOrPromise ) {
+	if ( urlOrPromise instanceof Promise ) {
+		return urlOrPromise;
+	}
+	return promiseForImage( imageForURL( urlOrPromise ) );
+};
 
```

Comment:
Worth a comment block here explaining the general approach and how promises are used?